modules/tracktion_engine/model/automation/tracktion_AutomatableParameter.cpp

```diff
@@ -317,11 +364,19 @@ struct MacroSource : public AutomationModifierSource
         const juce::ScopedLock sl (streamPositionLock);
         macro->updateFromAutomationSources (time);
         auto macroValue = macro->getCurrentValue();
+        
+        // BEATCONNECT MODIFICATION START
+        AutomationScaleHelpers::CurveType curveType;
 
-        const auto range = juce::Range<float>::between (assignment->inputLimitStart.get(), assignment->inputLimitEnd.get());
-        currentValue.store (AutomationScaleHelpers::mapValue (AutomationScaleHelpers::limitInputValue (macroValue, range),
-                                                              assignment->offset, assignment->value, assignment->curve),
-                            std::memory_order_release);
+        if (!assignment->state.hasProperty("curveType"))
```

Comment:
Where is "curveType" set? Somewhere externally?

Reply:
That's correct, it's set by right clicking on the curve control on the macro building and then selecting one of the four options. 
![image](https://github.com/BeatConnect/tracktion_engine/assets/133661475/a18c04b0-79ca-4972-b8d9-e82ec48f6878)

Reply:
OK. Just put "curveType" into IDs in tracktion_Identifiers.h with a BEATCONNECT MODIFICATION.
Maybe here:

![image](https://github.com/BeatConnect/tracktion_engine/assets/84411763/15f23b77-59db-4f20-bb12-94e2ee0cc02e)

I don't see how a tracktion modification is avoidable in this case. That said, have you fully investigated this?

Reply:
The two scenarios I considered for applying the macro curve were:
1) Do the math within a property change ahead of tracktion engine being aware of the curve, and then setting the curve value to 0.
2) Have plugins check their automatable parameters and redo the curve math there.

These approaches would leave tracktion engine unchanged, but also they would mean that the curves wouldn't be accounted for in the automation system, and that we would end up with two parallel systems for calculating curves.

I think these changes are made in the right place, where tracktion does all of its curved macro parameter value math; and minimal, where for 3 of the cases I was able to largely reuse existing code. It's not no changes to the engine but I feel they're reasonable.